src/test/app/AccountDelete_test.cpp

```diff
@@ -912,6 +912,275 @@ class AccountDelete_test : public beast::unit_test::suite
         env.close();
     }
 
+    void
+    testDestinationDepositAuthCredentials()
+    {
+        {
+            testcase(
+                "Destination Constraints with DepositAuth and Credentials");
+
+            using namespace test::jtx;
+
+            Account const alice{"alice"};
+            Account const becky{"becky"};
+            Account const carol{"carol"};
+
+            const char credType[] = "abcd";
+
+            Env env{*this};
+            env.fund(XRP(100000), alice, becky, carol);
+            env.close();
+
+            // carol issue credentials for becky
+            env(credentials::create(becky, carol, credType));
+            env.close();
+
+            // get credentials index
+            auto const jv =
+                credentials::ledgerEntryCredential(env, becky, carol, credType);
+            std::string const credIdx = jv[jss::result][jss::index].asString();
+
+            // Close enough ledgers to be able to delete becky's account.
+            incLgrSeqForAccDel(env, becky);
+
+            auto const acctDelFee{drops(env.current()->fees().increment)};
+
+            // becky use credentials but they aren't accepted
+            env(acctdelete(becky, alice),
+                credentials::IDs({credIdx}),
+                fee(acctDelFee),
+                ter(tecBAD_CREDENTIALS));
+            env.close();
+
+            {
+                // alice sets the lsfDepositAuth flag on her account.  This
+                // should prevent becky from deleting her account while using
+                // alice as the destination.
+                env(fset(alice, asfDepositAuth), fee(drops(10)));
+                env.close();
+            }
+
+            // Fail, credentials doesn’t belong to carol
+            env(acctdelete(carol, alice),
+                credentials::IDs({credIdx}),
+                fee(acctDelFee),
+                ter(tecBAD_CREDENTIALS));
+
+            // Fail, credentials not accepted
+            env(acctdelete(becky, alice),
+                credentials::IDs({credIdx}),
+                fee(acctDelFee),
+                ter(tecBAD_CREDENTIALS));
+            env.close();
+
+            // becky accept the credentials
+            env(credentials::accept(becky, carol, credType));
+            env.close();
+
+            // Fail, no depositPreauth
+            env(acctdelete(becky, alice),
+                credentials::IDs({credIdx}),
+                fee(acctDelFee),
+                ter(tecNO_PERMISSION));
+            env.close();
+
+            // alice create DepositPreauth Object
+            env(deposit::authCredentials(alice, {{carol, credType}}));
+            env.close();
+
+            // becky attempts to delete her account, but alice won't take her
+            // XRP, so the delete is blocked.
```

Comment:
Isn't this actually because Becky didn't provide credentials?

Reply:
Can you clarify a bit more in this comment?

Reply:
This exactly the same comment as from one of the previous DepositPreauth test and describe the same behavior for payment. I think it is clear enough.

Reply:
The previous DepositPreauth tests were about the accounts, here it's about the credentials.

Reply:
Ye, but the behaviors are the same in 2 test - you or allowed pay to alice  or not. It is just different parameters